en/components/grids_templates/cell-editing.md

```diff
@@ -180,6 +180,78 @@ This code is used in the sample below which implements an [`IgxSelectComponent`]
+
+Implementing this custom functionality can be done by utilizing the events of the grid. First we hook up to the grid's keydown events, and from here we can implement two functionalities:
+
+- Constant edit mode
+
+```typescript
+    public keydownHandler(event) {
+      const key = event.keyCode;
+      const grid = this.grid;
+      const activeElem = grid.navigation.activeNode;
+
+      if (
+        (key >= 48 && key <= 57) ||
+        (key >= 65 && key <= 90) ||
+        (key >= 97 && key <= 122)
+      ) {
+        // Number or Alphabet upper case or Alphabet lower case
+
+        const columnName = grid.getColumnByVisibleIndex(activeElem.column).field;
+        const cell = grid.getCellByColumn(activeElem.row, columnName);
+        if (cell && !grid.crudService.cellInEditMode) {
+          grid.crudService.enterEditMode(cell);
+          cell.editValue = event.key;
+        }
+      }
+```
+
+- `Enter`/ `Shift+Enter` navigation
+
+```typescript
+
+grid.crudService.enterEditMode(cell);
+
+// should we hit enter, we use the grid's navigation tools to get to the bottom cell instead
+if (key == 13)
+this.grid.navigateTo(nextRow, column, (obj) => {
+        obj.target.activate();
+        });
+
+// to find the next eiligible cell, we will use a custom method that will check the next suitable index
+  public getNextEditableRowIndex(currentRowIndex, dataView, previous){
+    //first we check if the currently selected cell is the first or the last
+    if (currentRowIndex < 0 || (currentRowIndex === 0 && previous) || (currentRowIndex >= dataView.length - 1 && !previous)) {
+        return currentRowIndex;
+    }
+    // in case using shift + enter combination, we look for the first suitable cell going up the field
+    if(previous){
+      return  dataView.findLastIndex((rec, index) => index < currentRowIndex && this.isEditableDataRecordAtIndex(index, dataView));
+    }
+    // or for the next one down the field
+    return dataView.findIndex((rec, index) => index > currentRowIndex && this.isEditableDataRecordAtIndex(index, dataView));
+  }
+```
+
+Please check the full sample bellow:
+
+## Angular Grid cell editing and edit templates Example
+
+<code-view style="height:650px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/grid/grid-editing-excel-style" alt="Angular Grid Excel Style Editing Example">
+</code-view>
+
+
+Main benefits of the above approach include:
+
+- Constant edit mode: typing while a cell is selected will immediately enter it in edit mode with the value typed, replacing the existing one
+- Any non-data rows are skipped when navigating with `Enter`/`Shift+Enter`
+- Should grouping, master detail or any other feature adding headers as records is enabled, enter will navigate you to the next editable cell skipping all headers. Cycling quickly trough the values has neever been easier
```

Comment:
Arent the last two with the same meaning. I think that they should be merged somehow. Also, the last part `Cycling quickly trough the values has neever been easier` has a typo and IMO it is not appropriate for the topics.